internal/tokens/https/https.go

```diff
@@ -0,0 +1,223 @@
+	}
+
+	h.client = restclient.New(h.ctx, prefix)
+	h.capabilities = &tokens.Capabilities{}
+
+	if prefix.GetString(wsclient.WSConfigKeyPath) == "" {
+		prefix.Set(wsclient.WSConfigKeyPath, "/api/ws")
+	}
+	h.wsconn, err = wsclient.New(ctx, prefix, nil)
+	if err != nil {
+		return err
+	}
+
+	h.closed = make(chan struct{})
+	go h.eventLoop()
+
+	return nil
+}
+
+func (h *HTTPS) Start() error {
+	return h.wsconn.Connect()
+}
+
+func (h *HTTPS) Capabilities() *tokens.Capabilities {
+	return h.capabilities
+}
+
+func (h *HTTPS) handleReceipt(ctx context.Context, data fftypes.JSONObject) error {
+	l := log.L(ctx)
+
+	requestID := data.GetString("id")
+	success := data.GetBool("success")
+	message := data.GetString("message")
+	if requestID == "" {
+		l.Errorf("Reply cannot be processed: %+v", data)
+		return nil // Swallow this and move on
+	}
+	updateType := fftypes.OpStatusSucceeded
+	if !success {
+		updateType = fftypes.OpStatusFailed
+	}
+	l.Infof("Tokens '%s' reply (request=%s) %s", updateType, requestID, message)
+	return h.callbacks.TokensTxUpdate(requestID, updateType, message, data)
+}
+
+func (h *HTTPS) handleTokenPoolCreate(ctx context.Context, data fftypes.JSONObject) (err error) {
+	ns := data.GetString("namespace")
+	name := data.GetString("name")
+	clientID := data.GetString("client_id")
+	tokenType := data.GetString("type")
+	poolID := data.GetString("pool_id")
+	authorAddress := data.GetString("author")
+
+	if ns == "" ||
+		name == "" ||
+		clientID == "" ||
+		tokenType == "" ||
+		poolID == "" ||
+		authorAddress == "" {
+		log.L(ctx).Errorf("TokenPool event is not valid - missing data: %+v", data)
+		return nil // move on
+	}
+
+	uuid, err := fftypes.ParseUUID(ctx, clientID)
+	if err != nil {
+		log.L(ctx).Errorf("TokenPool event is not valid - bad uuid (%s): %+v", err, data)
+		return nil // move on
+	}
+
+	pool := &fftypes.TokenPool{
+		ID:        uuid,
+		Namespace: ns,
+		Name:      name,
+		Type:      fftypes.LowerCasedType(tokenType),
+		PoolID:    poolID,
+	}
+
+	// If there's an error dispatching the event, we must return the error and shutdown
+	return h.callbacks.TokenPoolCreated(pool, authorAddress, data)
+}
+
+func (h *HTTPS) eventLoop() {
+	defer close(h.closed)
+	l := log.L(h.ctx).WithField("role", "event-loop")
+	ctx := log.WithLogger(h.ctx, l)
+	for {
+		select {
+		case <-ctx.Done():
+			l.Debugf("Event loop exiting (context cancelled)")
+			return
+		case msgBytes, ok := <-h.wsconn.Receive():
+			if !ok {
+				l.Debugf("Event loop exiting (receive channel closed)")
+				return
+			}
+
+			var msg wsEvent
+			err := json.Unmarshal(msgBytes, &msg)
+			if err != nil {
+				l.Errorf("Message cannot be parsed as JSON: %s\n%s", err, string(msgBytes))
+				continue // Swallow this and move on
+			}
+			l.Debugf("Received %s event %s", msg.Event, msg.ID)
+			switch msg.Event {
+			case messageReceipt:
+				err = h.handleReceipt(ctx, msg.Data)
```

Comment:
Note: we get a receipt here from the Tokens microservice, which is actually wrapped from the receipt it received from ethconnect.

Because we're also directly subscribed to ethconnect, the blockchain plugin _also_ gets the ethconnect receipt, dumps a bunch of logs about how it's unrecognized, and then moves on. It all works alright, but need a follow-up to clean up that logging behavior (without breaking ethconnect receipts that we do care about).

Reply:
Fixed in #156